### PR TITLE
LPD-23797 Upgrade to Java Zulu21

### PR DESCRIPTION
**Pull request by Wykenz**
Hi @natocesarrego I made all jdk21 related files separately from the jdk11.  I don't think this will be the final form of this PR but I wanted to show you the code, now the image works but I'm a little lost on the first part of the ticket description, for example: 
"Only 7.4 Liferay Portal Docker images should be affected, the Docker images associated to the previous Liferay Portal releases, like 7.3, shouldn’t be affected."
I'm not familiar where should I  set these docker image versions. Can you help me with that?

### CODE REVIEW

File: templates/jdk11/Dockerfile [hunk 1]
```diff
@@ -41,6 +41,7 @@ RUN apt-get update && \
 	apt-get install --no-install-recommends --yes jattach && \
 	apt-get upgrade --yes && \
 	apt-get clean && \
+	echo "${TARGETARCH} ${LABEL_ZULU_11_VERSION}" && \
```
Comment (natocesarrego) [inline templates/jdk11/Dockerfile:44]: @Wykenz, why did you need to set this?

File: templates/jdk21/resources/home/liferay/.bashrc [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (natocesarrego) [inline templates/jdk21/resources/home/liferay/.bashrc:1]: @Wykenz, can't we reuse this same file presented in the folder `templates/jdk11/resources`?

File: templates/jdk21/resources/usr/local/bin/set_java_version.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+function create_symlink {
```
Comment (natocesarrego) [inline templates/jdk21/resources/usr/local/bin/set_java_version.sh:3]: @Wykenz, can't we reuse this same file presented in the folder `templates/jdk11/resources`

File: templates/jdk11/Dockerfile (hunk 1 above)
Comment (Wykenz) [reply templates/jdk11/Dockerfile:44]: oh  sorry it just stayed there by mistake, I only used that line for testing and I will remove it

File: templates/jdk21/resources/home/liferay/.bashrc (hunk 2 above)
Comment (Wykenz) [reply templates/jdk21/resources/home/liferay/.bashrc:1]: @natocesarrego actually I don't know if we want to keep the jdk11 folder or not because in the ticket there is this line "Lastly, probably the [code](https://github.com/liferay/liferay-docker/blob/master/templates/jdk11/Dockerfile) should be modified to implement this requested change."  Here "modified" means update the existing jdk11 files to jdk21 or have both in the repository? That is why I made it next to the existing one so if we need only the jdk21 then I can delete the jdk11 folder, but if we need both then I change the jdk21 according to it.

Comment (natocesarrego) [reply templates/jdk21/resources/home/liferay/.bashrc:1]: Let's keep both for now

File: templates/jdk21/resources/usr/local/bin/set_java_version.sh (hunk 3 above)
Comment (Wykenz) [reply templates/jdk21/resources/usr/local/bin/set_java_version.sh:3]: Hi @natocesarrego, I tried to do this but as I can see the building process has a pattern on how the files can be made and used.
Every image have a build script, for zulu21 I made [build_jdk21_image.sh](https://github.com/Wykenz/liferay-docker/blob/LPD-23797-generate-docker-images-using-java-zulu-21/build_jdk21_image.sh ) based on the jdk11's script. There is this function [make_temp_directory](https://github.com/Wykenz/liferay-docker/blob/LPD-23797-generate-docker-images-using-java-zulu-21/build_jdk21_image.sh#L48) which is from [_common.sh](https://github.com/Wykenz/liferay-docker/blob/LPD-23797-generate-docker-images-using-java-zulu-21/_common.sh#L184) that creates a tempdir with all the contents of the image, and this function is used by every build_image.sh files. The other images have its resources under their own directory so I feel like separating them would go against the standard.

Comment (natocesarrego) [reply templates/jdk21/resources/usr/local/bin/set_java_version.sh:3]: Hi @Wykenz, you can do some little changes, as highlighted in red in the image below, to reuse the code, please see:

image

Can you try to move on like this?
